plugins/processors/awsappsignals/rules/keeper.go

```diff
@@ -3,15 +3,21 @@
 
 package rules
 
-import "go.opentelemetry.io/collector/pdata/pcommon"
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+
+	"github.com/aws/amazon-cloudwatch-agent/plugins/processors/awsappsignals/common"
+)
 
 type KeepActions struct {
-	Actions []ActionItem
+	Actions                 []ActionItem
+	markDataPointAsReserved bool
```

Comment:
Can you remind me what `markDataPointAsReserved` does? 

I see that we pass in `!limiterConfig.Disabled` to this value... not sure I completely understand

Reply:
Yeah, it is to make sure the metrics that `Replaced` or `Kept` by the custom rules will not be aggregated by MetricLimiter which happens later.  

So if MetricLimiter is enabled, we will inject an placeholder `markDataPointAsReserved` attribute to Metrics, and the MetricLimiter will not process such metrics later. If MetricLimiter is not enabled in CWAgent, we just don't do anything.